OvmfPkg/QemuKernelLoaderFsDxe/QemuKernelLoaderFsDxe.c

```diff
@@ -942,6 +944,33 @@ FetchBlob (
   UINTN   Idx;
   UINT8   *ChunkData;
 
+  if (Blob->FwCfgName) {

```

Comment:
Can we integrate this a bit more nicely please?

- Local variables should be declared in the parent scope
- There is some duplication of code, and inconsistency between the versions (e.g., whether a DEBUG error message is printed when failing to allocate Blob data
- The chunking logic is omitted for named BLOBs - is there a reason for that?
- Success handling anti-pattern is used.

Reply:
>     * Local variables should be declared in the parent scope

Will fix.

>     * There is some duplication of code, and inconsistency between the versions (e.g., whether a DEBUG error message is printed when failing to allocate Blob data

Can add the error msg.  The minor duplication is hard to remove witout making the control flow more complex though.

>     * The chunking logic is omitted for named BLOBs - is there a reason for that?

It's not needed.  The kernel is delivered by qemu in two chunks (the setup header separately, patched by qemu for direct kernel load in bios mode).  `etc/boot/kernel` is not chunked and not modified.

>     * Success handling anti-pattern is used.

`--verbose` please.  Do you mean the early `return EFI:SUCCESS` ?

Reply:
I mean doing

```
Status = Foo ();
if (Status == EFI_SUCCESS) {
  ...
}
```